pkg/reconcile/reconcile.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcile
+
+import (
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Result contains the result of a Reconcile invocation.
+type Result struct {
+	// Requeue tells the Controller to requeue the reconcile key.  Defaults to false.
+	Requeue bool
```

Comment:
One question (not scope of `reconcile` pkg though). What is the semantics for handling the error returned by the `Reconcile()`. Is it going to be simply logging the error and ignoring it ? or using it for re-queuing ? If it is the former, then current controller writer might get surprised a bit (not saying it is a bad thing though).

Reply:
the latter. will document